enhancements/authentication/direct-external-oidc-provider.md

```diff
@@ -212,6 +212,90 @@ These changes have already been implemented, and the initial PR for them can be
 
+#### authorization.openshift.io/RestrictSubjectBindings admission plugin considerations
+
+OpenShift's default admission plugin `authorization.openshift.io/RestrictSubjectBindings`
+utilizes the OAuth stack, specifically the `User` and `Group` APIs, to ensure `RoleBinding`
+subjects do not violate any restrictions in the namespace they are going to be created in.
+These restrictions are specified by `RoleBindingRestriction` resources that are served
+by the openshift-apiserver.
+
+Removing the OAuth stack entirely during external OIDC setup may result in the `authorization.openshift.io/RestrictSubjectBindings` admission plugin not functioning as expected.
+
+Anticipated impacts are:
+
+- Logs in the KAS reflecting that informers for OAuth APIs could not be started
+- Logs in the KAS reflecting that the oauth-apiserver is unreachable
+- Errors, that are not actionable from the perspective of users, occurring during evaluation of `RoleBinding` resources when a `RoleBindingRestriction` resource in the same namespace specifies user/group restrictions
+
+These impacts may cause concern for cluster administrators that their cluster is in an unhealthy state and may not be easily correlated to the enablement of the OIDC functionality.
+
+To ensure a holistic change in cluster behavior when enabling the OIDC functionality, and reduce potentially concerning behavior in the control plane components, it is proposed that changes are made to:
+
+- The `authorization.openshift.io/RestrictSubjectBindings` admission plugin to account for necessary behavior changes
+- The openshift-apiserver to reject `RoleBindingRestriction` resources that specify user/group restrictions when OIDC functionality is enabled
+- The `Authentication` api to communicate when OIDC can't be enabled due to existing `RoleBindingRestriction` resources that specify user/group restrictions
+
+##### Changes to the kube-apiserver
+
+To account for the impacts to the `authorization.openshift.io/RestrictSubjectBindings` admission plugin, the OpenShift-specific patch to the kube-apiserver that adds this admission plugin will be updated such that:
+
+- Informers for the `Group` API are not started if the `Authentication` resource `.Spec.Type` is set to `OIDC`
+- The post-start hook that checks for oauth-apiserver connectivity will be skipped if the `Authentication` resource `.Spec.Type` is set to `OIDC`
+- `RoleBinding`s will be rejected if there exists a `RoleBindingRestriction` that specifies user and/or group restrictions in the namespace the `RoleBinding` is being created
```

Comment:
Let's discuss this further -- the RBRs have been built with OAuth in mind, but AFAIU RoleBindings should still work with OIDC users/groups.